### ISSUE TITLE
ArangoDB 3.2beta2 unexpected "unique constraint violated (while executing)" 

### ISSUE THREAD

**Issue by DeShadow** [arangodb/arangodb#2652]
ArangoDB 3.2beta2

I created empty database and imported by arangorestore dump of database. After that I try to create new document by AQL:
```
await arango.queryOne(`insert MERGE(@authLogData, { date: DATE_ISO8601(DATE_NOW()) }) in authLogs`, { authLogData });
```

authLogData is
```
{
  "userId": "167",
  "type": "token",
  "isSuccess": true
}
```

I have no `_key` field. Collection `authLogs` has only one default `_key` primary index. And sometimes (in 90% of cases) I have error:
```
409
1210
"unique constraint violated (while executing)"
"ArangoError"
```

It's very strange...

**Comment by DeShadow** [comment]
UPDATE:
When I tested the same db-directory with beta1, there is no this bug.

**Comment by jsteemann** [comment]
What storage engine have you been using? mmfiles or rocksdb?

**Comment by DeShadow** [comment]
@jsteemann RocksDB 👍

**Comment by jsteemann** [comment]
Is there any special key generator set up for that collection?

**Comment by DeShadow** [comment]
No. Simple collection without any additional properties.
From dump .structure.json
```
{"indexes":[],"parameters":{"allowUserKeys":true,"cid":"448207159","count":0,"deleted":false,"doCompact":true,"id":"448207159","indexBuckets":8,"indexes":[{"fields":["_key"],"id":"0","selectivityEstimate":1,"sparse":false,"type":"primary","unique":true}],"isSmart":false,"isSystem":false,"isVolatile":false,"journalSize":33554432,"keyOptions":{"allowUserKeys":true,"type":"traditional"},"name":"authLogs","numberOfShards":1,"planId":"448207159","replicationFactor":1,"shardKeys":["_key"],"shards":{},"status":4,"type":2,"version":6,"waitForSync":false}}
```

**Comment by jsteemann** [comment]
I have hope this will also be fixed by https://github.com/arangodb/arangodb/issues/2651, because that PR will fix not only the persistence the last assigned auto-increment key value, but should also lead to the same key not being reassigned with other key generators.

**Comment by jsteemann** [comment]
Should be fixed in 3.2beta3.

**Comment by dothebart** [comment]
We're pleased to anounce that ArangoDB 3.2 containing this is now available for download - Closing.
